dispatches/models/renewables_case/wind_power.py

```diff
@@ -151,12 +151,25 @@ def setup_resource(self):
                 if abs(sum(r[2] for r in resource) - 1) > 1e-3:
                     raise ValueError("Error in 'resource_probability_density' for time {}: Probabilities of "
                                      "Wind Speed and Direction Probability Density Function must sum to 1")
-                self.wind_simulation.Resource.wind_resource_distribution = resource
-                self.wind_simulation.execute(0)
-                self.capacity_factor[time].set_value(self.wind_simulation.Outputs.capacity_factor / 100.)
-
-            @self.Constraint(self.flowsheet().config.time)
-            def elec_from_capacity_factor(b, t):
-                return b.electricity[t] == self.system_capacity * self.capacity_factor[t]
+                if len(resource) != 1:
+                    raise NotImplementedError
+                wind_simulation = self.setup_atb_turbine()
+                wind_simulation.Resource.weibull_wind_speed = resource[0][0]
+                wind_simulation.execute(0)
+                self.capacity_factor[time].set_value(wind_simulation.Outputs.capacity_factor / 100.)
         else:
             raise ValueError("Config with 'resource_probability_density' must be provided using `default` argument")
+
+    def initialize(self, state_args={}, state_vars_fixed=False,
+                   hold_state=False, outlvl=idaeslog.NOTSET,
+                   temperature_bounds=(260, 616),
+                   solver=None, optarg=None):
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="properties")
+        solve_log = idaeslog.getSolveLogger(self.name, outlvl,
+                                            tag="properties")
```

Comment:
If the model has a single equation, then we don't have to use ipopt. We can just "set" the value of electricity from the equation:
```python
for t in self.flowsheet().config.time:
    self.electricity[t].value = self.system_capacity.value * self.capacity_factor[t]
```
This would save some computational time, especially if you need to call initialize method multiple times in the code.